src/Discord.Net.Core/CDN.cs

```diff
@@ -12,26 +12,45 @@ public static class CDN
         /// </summary>
         /// <param name="appId">The application identifier.</param>
         /// <param name="iconId">The icon identifier.</param>
+        /// <param name="format">The format to return. Mustn't be a gif.</param>
+        /// <param name="size">The size of the image to return in. This can be any power of two between 16 and 2048.</param>
         /// <returns>
         ///     A URL pointing to the application's icon.
         /// </returns>
-        public static string GetApplicationIconUrl(ulong appId, string iconId)
-            => iconId != null ? $"{DiscordConfig.CDNUrl}app-icons/{appId}/{iconId}.jpg" : null;
+        public static string GetApplicationIconUrl(ulong appId, string iconId, ImageFormat format, ushort size)
+        {
+            if (string.IsNullOrWhiteSpace(iconId))
+                return null;
+            if (format == ImageFormat.Gif)
+                throw new ArgumentException("Requested image format mustn't be a gif.");
```

Comment:
```suggestion
                throw new ArgumentException(message: "Requested image format mustn't be a gif.", paramName: nameof(format));
```
I think we should include `paramName` in `ArgumentException`s when possible. I'm using named parameters because I know that these parameters are sometimes switched around.